e2e/e2e_test.go

```diff
@@ -27,6 +27,7 @@ func init() {
 	flag.StringVar(&upgradeVersion, "upgrade-version", "v3.3.1", "target version for upgrade testing")
 	flag.StringVar(&cephCSINamespace, "cephcsi-namespace", defaultNs, "namespace in which cephcsi deployed")
 	flag.StringVar(&rookNamespace, "rook-namespace", "rook-ceph", "namespace in which rook is deployed")
+	flag.StringVar(&tenantNamespace, "tenant-namespace", "tenant", "namespace for tenant based testing")
```

Comment:
@Rakshith-R @nixpanic do we need a new namespace? currently, we have `cephCSINamespace` and a unique name space where test cases run.

Reply:
As already mentioned, 
https://github.com/ceph/ceph-csi/pull/2239#discussion_r664473726
> The aim of this particular test case is to test if secret can be fetched from user specified `secretName` & `secretNamespace`.
> ( user specified `secretName` & tenant/k8s namespace of PVC is tested in the other e2e)
> 
> `cephcsiNamespace == default` but default is replaced with e2e testing unique namespace in `kms-config.yaml`, that's why I used `rookNamespace` [where secret will be created(then destroyed) in the same function being referred for testing] .

`"default"` is replaced by `cephcsiNamesphace` during vault deployment in `kms-config` & other files. That's why we're creating and using a new `tenant` namespace.

Reply:
Apologies :sweat_smile: , I get it `cephcsiNamespace` should work. Reverted the new namespace and using `cephcsiNamespace` now. 
[since default will be replaced by `cephcsiNamespace` and PVC will be created in `f.Unique` namespace. my bad :facepalm: ]

Reply:
np. lets hear from Niels also